test/emulation.spec.js

```diff
@@ -351,4 +351,65 @@ describe('Emulation', () => {
       expect(error.message).toBe('Invalid timezone ID: Baz/Qux');
     });
   });
+
+  describeFailsFirefox('Page.emulateVisionDeficiency', function () {
+    it('should work', async () => {
+      const { page, server } = getTestState();
+
+      await page.setViewport({ width: 500, height: 500 });
+      await page.goto(server.PREFIX + '/grid.html');
+
+      {
+        await page.emulateVisionDeficiency('none');
+        const screenshot = await page.screenshot();
+        expect(screenshot).toBeGolden('screenshot-sanity.png');
+      }
+
+      {
+        await page.emulateVisionDeficiency('achromatopsia');
+        const screenshot = await page.screenshot();
+        expect(screenshot).toBeGolden('vision-deficiency-achromatopsia.png');
+      }
+
+      {
+        await page.emulateVisionDeficiency('blurredVision');
+        const screenshot = await page.screenshot();
+        expect(screenshot).toBeGolden('vision-deficiency-blurredVision.png');
+      }
+
+      {
+        await page.emulateVisionDeficiency('deuteranopia');
+        const screenshot = await page.screenshot();
+        expect(screenshot).toBeGolden('vision-deficiency-deuteranopia.png');
+      }
+
+      {
+        await page.emulateVisionDeficiency('protanopia');
+        const screenshot = await page.screenshot();
+        expect(screenshot).toBeGolden('vision-deficiency-protanopia.png');
+      }
+
+      {
+        await page.emulateVisionDeficiency('tritanopia');
+        const screenshot = await page.screenshot();
+        expect(screenshot).toBeGolden('vision-deficiency-tritanopia.png');
+      }
+
+      {
+        await page.emulateVisionDeficiency('none');
+        const screenshot = await page.screenshot();
+        expect(screenshot).toBeGolden('screenshot-sanity.png');
+      }
+    });
```

Comment:
nit: I'd personally have one individual `it` for each test rather than the blocks (I think it leads to nicer errors if a test fails) but take it or leave it

Reply:
I see what you're saying. I think the reason it's done this way for existing tests is because we'd otherwise have to duplicate the `page.setViewport` + `page.goto` setup for each subtest.